### DISCUSSION TITLE
Get script level metrics started with php

### DISCUSSION THREAD

**Discussion by xxm404** [netdata/netdata#15246] [Q&A] [answered]
I want to get script level metrics started with php，start cmd like this
```
php abc.php
```
After I installed netdata, I opened the apps.plugin, I found that the following configuration exists in apps_groups.conf
```
# web/ftp servers

httpd: apache* httpd nginx* lighttpd hiawatha caddy h2o
proxy: squid* c-icap squidGuard varnish*
php: php* lsphp*
ftpd: proftpd in.tftpd vsftpd
uwsgi: uwsgi
unicorn: *unicorn*
puma: *puma*
```
I have a lot of php processes running on my machine, using `ps -e | grep php | wc -l `the result is 160, but I don't find any php process level indicators on netdata's Kanban board, how can I modify the configuration?

**Comment by andrewm4894** [comment]
Hi @xxm404

You would need to edit the `/etc/netdata/apps_groups.conf` file to add your own customizations. 

Something like below:

```
$ cd /etc/netdata
$ ./edit-config apps_groups.conf
```

![image](https://github.com/netdata/netdata/assets/2178292/95427f7c-d559-414b-8ea3-56eb9628d701)

save your edits and then [restart netdata](https://learn.netdata.cloud/docs/maintaining/starting-and-stopping-netdata-agents)

[This chatgpt](https://chat.openai.com/share/c0c2a955-ffd6-4053-b77a-3b792b641634) explains it well enough too. 

Try it and let us know how you get on.

**Reply by andrewm4894** [reply]
one small note: i put `abc` on the line above `php` so that it will be matched first (line order matters in this conf file) and excluded from the `php` group.

is mentioned in here: https://learn.netdata.cloud/docs/data-collection/apm/application-monitoring#configuration

**Reply by xxm404** [reply]
@andrewm4894 Thank you for your reply，I modified the configuration as you suggested, and restarted netdata, but I still can't find any indicators I want in dashboard
<img width="682" alt="image" src="https://github.com/netdata/netdata/assets/46340314/a06a1fa2-0253-4484-a859-84dd32227ed5">
dashboard
<img width="1144" alt="image" src="https://github.com/netdata/netdata/assets/46340314/186dac89-b901-4516-a8dd-c9e1bc592751">

**Reply by andrewm4894** [reply]
Maybe try:

```
runtest: run_test.php*
```

How does it look when you run `ps -e | grep php` ?

**Reply by andrewm4894** [reply]
Oh @xxm404 it would not be on that chart btw - it would be under the Applications section. 

![image](https://github.com/netdata/netdata/assets/2178292/f8e2e8af-6481-48ba-ac99-632786874f6e)

**Reply by xxm404** [reply]
@andrewm4894 
On my web page running locally, I don't find the application section,Do I need to modify any configuration?
<img width="183" alt="image" src="https://github.com/netdata/netdata/assets/46340314/9a36ebb4-5868-4365-b046-aaf5aeb75bcb">
When i run `ps -e | grep php` get this
<img width="504" alt="image" src="https://github.com/netdata/netdata/assets/46340314/45244d4c-b813-480b-b4ce-ae100ca2098a">

**Reply by andrewm4894** [reply]
you might need to check if the apps plugin is enabled or disabled - it should be enabled by default: 

https://learn.netdata.cloud/docs/configuring/how-to-optimize-the-netdata-agent-s-performance#disable-unneeded-plugins-or-collectors

What netdata version are you running and how did you install it?

**Reply by xxm404** [reply]
> What netdata version are you running and how did you install it?

I install netdata by `packaging/makeself/build-x86_64-static.sh` ang get this `netdata-x86_64-v1.39.0-118-g560d5c856.gz.run`

> you might need to check if the apps plugin is enabled or disabled - it should be enabled by default:

I think my app plugin is enabled, the config file like this
<img width="544" alt="image" src="https://github.com/netdata/netdata/assets/46340314/350fdf9f-20b0-4705-b8bb-9f121f473298">

**Reply by xxm404** [reply]
when i run `systemctl status netdata` i get one log 
```
[13146]: running: exec /opt/netdata/usr/libexec/netdata/plugins.d/cgroup-network-helper.sh --cgroup '/sys/fs/cgroup/cpu,cpuacct/docker/9b27d9baa0e24293f80bb3fa1a91799a8680f284de128b6963aca0b6fee7c341'
[13146]: child pid 13147 exited with code 1.
```

**Reply by andrewm4894** [reply]
What os are you installing on?

**Reply by andrewm4894** [reply]
Strange to see the apps section missing.

@ilyam8 do you know if there is a way to run apps plugins in debug mode?

**Reply by xxm404** [reply]
> What os are you installing on?

Ubuntu 18.04.4 LTS

**Reply by xxm404** [reply]
@andrewm4894 Hi，maybe you can tell me how to debug it.

**Comment by xxm404** [comment]
Can someone help me

**Comment by ilyam8** [comment]
Hey. Find the PID of the process you are interested in (`ps aux | grep <PATTERN>`) and show

```bash
cat /proc/PID/comm
cat /proc/PID/cmdline | strings
```

**Reply by xxm404** [reply]
```
cat /proc/3335/comm 
php

cat /proc/3335/cmdline | strings
/usr/local/php72/bin/php
/home/work/env/php/etc/php.ini
run_test.php
```

**Reply by ilyam8** [reply]
I think you should have `php` dimension in the "Applications" chart. `php: php* lsphp*` should match all processes with `php` comm.

You can find out all the processes (PID) that are aggregated into the `php` group using debug mode

```bash
cd /usr/libexec/netdata/plugins.d
sudo ./apps.plugin debug without-users without-groups 2>&1 | grep -m 1 "target 'php' has aggregated"
```

if there are any processes this command will print them and exit.

**Reply by xxm404** [reply]
> sudo ./apps.plugin debug without-users without-groups 2>&1 | grep -m 1 "target 'php' has aggregated". 
```
apps.plugin: target 'php' has aggregated 162 processes: 32412 32326 32282 31475 31344 30851 30698 30685 30660 30647 30437 30107 29926 29892 29865 29742 29695 29557 29546 29530 29513 29477 29102 28556 28554 28293 28227 28181 28166 28090 28088 28087 28086 28056 28049 27707 27058 26773 26127 25932 25890 25869 25843 24843 24442 24405 24318 23831 23780 21500 20006 19985 19382 18228 16291 16287 16149 16145 16052 16040 15959 15952 15875 15867 15659 15653 15449 15432 15375 15371 15212 15204 15115 15105 15018 15010 14890 14884 14780 14772 14666 14659 14543 14533 14422 14414 14316 14305 12379 12378 11424 11420 9659 9645 7044 7038 6977 6959 6931 6914 6844 6819 6815 6802 6798 6567 6559 6529 6525 6476 6472 6462 6458 6449 6445 6436 6432 6422 6418 6409 6405 6396 6392 6383 6379 6369 6365 6314 6310 6306 6302 6277 6269 6227 6218 6203 6199 6188 6184 6042 6038 6007 6003 5994 5990 5667 5607 5564 4244 4177 3934 3928 3862 3860 3859 3824 3808 2199 726 436 314 313
```
pid 15375 is my interested process but in my netdata webui still can not found application section
<img width="205" alt="image" src="https://github.com/netdata/netdata/assets/46340314/86042550-ff31-428c-b703-22d9991c46c0">

**Reply by ilyam8** [reply, marked as answer]
Ah, it is a different issue. I think you are using Netdata stable release, right? There is a bug and some external plugins don't work (apps.plugin, the one that collects Applications metrics is among them). We will release v1.40.1 that fixes the problem soon.

**Reply by xxm404** [reply]
Thanks a lot, looking forward to this update.

**Reply by ilyam8** [reply]
v1.40.1 was released yesterday.
